packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketHistoryList.tsx

```diff
@@ -75,9 +75,11 @@ export function RedPacketHistoryList(props: RedPacketHistoryListProps) {
     return (
         <div className={classes.root}>
             <List style={{ padding: '16px 0 0' }}>
-                {histories.map((history, i) => (
-                    <RedPacketInHistoryList key={i} history={history} onSelect={onSelect} />
-                ))}
+                {histories
+                    .filter((x) => x.chainId === chainId)
```

Comment:
Filtering in UI rather than in hook. The reason is after switch to a new chain, the hook still returns the old chain histories data for a while which causes the UI lag when data is huge. Although I set the `chainId` as the dependency for the hook.

Reply:
UPDATES must come from the alteration of histories. I think we should investigate the reason for the laggy.